File: core/core-api/src/main/java/com/coffee/gu/cancel/CancelManager.java
package com.coffee.gu.cancel;

import com.coffee.gu.payment.Payment;

import org.springframework.stereotype.Component;

@Component
public class CancelManager {

    private final CancelRepository cancelRepository;

    public CancelManager(CancelRepository cancelRepository) {
        this.cancelRepository = cancelRepository;
    }

    public Cancel cancel(Payment payment, String externalCancelKey) {
        return cancelRepository.cancel(payment, externalCancelKey);
    }
}

File: core/core-api/src/main/kotlin/com/coffee/gu/cancel/CancelManager.kt
package com.coffee.gu.cancel

import com.coffee.gu.payment.Payment
import org.springframework.stereotype.Component

@Component
class CancelManager(
    private val cancelRepository: CancelRepository,
) {
    fun cancel(payment: Payment, externalCancelKey: String): Cancel {
        return cancelRepository.cancel(payment, externalCancelKey)
    }
}
